### PR TITLE
API: do not de-reference foreign Keys 

### PR DESCRIPTION
**Pull request by tacaswell**

### CODE REVIEW

**Comment by danielballan** [comment]
THS PR HS A RLY CRYP. TLT.

File: databroker/headersource/base.py [hunk 1]
```diff
@@ -7,10 +7,7 @@
 
 
 class MDSROTemplate(object):
-    def __init__(self, config, version=1, auth=False):
-        if auth:
-            raise NotImplementedError("This implementation of MDS does not "
-                                      "support authentication.")
+    def __init__(self, config, version=1):
```
Comment (danielballan) [inline databroker/headersource/base.py]: While you're here, do you want to [move version into the config dict](https://github.com/NSLS-II/databroker/issues/185)?

File: databroker/headersource/base.py [hunk 2]
```diff
@@ -1,8 +1,16 @@
 from __future__ import (absolute_import, division, print_function,
                         unicode_literals)
 from . import core
+import numpy as np
 
 
+def _sanitize_np(val):
```
Comment (danielballan) [inline databroker/headersource/base.py]: In #195 (awaiting fixes before merge) I add `sanitize_np` a public function in utils. Leaving this here is fine for this PR, but removing the leading underscore from the name would make the rebase easier. Optional!

Comment (tacaswell) [reply databroker/headersource/base.py]: There is probably a longer discussion to be had about where this should be done (would be better to push lower for ex the hdf5 backend?

Comment (danielballan) [reply databroker/headersource/base.py]: It's possible that this is actually safe to remove once #195 is in, which does handle it lower at least in places. Let's revisit there. If you don't want to delete the underscore, that's fine, just trying to save 5 seconds of hassle while I have just burned it here. :- )

**Comment by tacaswell** [comment]
Both done.

**Comment by danielballan** [comment]
:clap:

**Pull request merged**